### PR TITLE
timers plugin: Use item and sprite IDs for timers

### PR DESCRIPTION
**Pull request by Nightfirecat**

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/GameTimer.java [hunk 1]
```diff
@@ -92,28 +99,69 @@
 		this.duration = Duration.of(time, unit);
 	}
 
+	GameTimer(int itemId, int quantity, String description, Integer graphicId, long time, ChronoUnit unit)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/GameTimer.java]: shouldnt this quantity be just always 1?

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/GameTimer.java [hunk 2]
```diff
@@ -92,28 +99,69 @@
...
+		this.spriteId = spriteId;
+	}
+
 	GameTimer(String imageResource, String description, long time, ChronoUnit unit)
 	{
 		this(imageResource, description, null, time, unit);
 	}
 
-	public BufferedImage getImage()
+	GameTimer(int itemId, int quantity, String description, long time, ChronoUnit unit)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/GameTimer.java]: See my previous comment. If you need to differentiate i would use some enum or something probably

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimerTimer.java [hunk 3]
```diff
@@ -33,9 +35,12 @@
 {
 	private final GameTimer timer;
 
+	static ItemManager itemManager;
+	static SpriteManager spriteManager;
+
 	public TimerTimer(GameTimer timer, Plugin plugin)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TimerTimer.java]: plugin.getItemManager and plugin.getSpriteManager. And remove the static assigns. or even send both to constructor

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/GameTimer.java (hunk 1 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/GameTimer.java]: I guess? I wrote this constructor with two reasons in mind:

1. The item and sprite constructors can't both have the same arguments (if there's a better way to handle that, I'm all ears)
2. It's unlikely, but possible someone might want to have a timer with a stackable item as the displayed image, like a pile of coins.

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/GameTimer.java (hunk 2 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/GameTimer.java]: It's enums all the way down!

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimerTimer.java (hunk 3 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TimerTimer.java]: I went ahead with the constructor approach.

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/GameTimerImageType.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.timers;
+
+enum GameTimerImageType
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/GameTimerImageType.java]: I feel like this whole enum isn't needed.

**Pull request merged**